src/main/java/com/redhat/cloud/custompolicies/app/rest/SettingsService.java

```diff
@@ -57,43 +57,64 @@
 
   @Operation(summary = "Save or update settings from the settings UI")
   @APIResponse(responseCode = "200", description = "Saving was ok")
+  @APIResponse(responseCode = "403", description = "User has no permission to change settings")
+  @APIResponse(responseCode = "500", description = "Saving of settings failed")
   @POST
   @Path("/")
   @Transactional
   public Response saveSettings(@Valid SettingsValues values) {
 
+    Response.ResponseBuilder builder;
+
+    if (!user.canWriteAll()) {
+      return Response.status(Response.Status.FORBIDDEN).entity("You don't have permission to change settings").type("text/plain").build();
+    }
+
     values.username = user.getName();
     values.accountId = user.getAccount();
     SettingsValues tmp = SettingsValues.findById(user.getName());
     if (tmp != null) {
       tmp.immediateEmail = values.immediateEmail;
       tmp.dailyEmail = values.dailyEmail;
     } else {
-      values.persistAndFlush();
+      values.persist();
```

Comment:
This still persists changes even if the store fails?

Reply:
Should not, as there is a `@Transactional` annotation

Reply:
Line 97 has ``catch (Exception e) {`` thus the JTA doesn't get any exception. So, what would rollback?

Reply:
You are right. I changed this now.